tools/sort_actor_funcs.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("file", type=str, help="Path to c file to sort")
+args = parser.parse_args()
+
+file_path = Path(args.file).resolve()
+
+if not file_path.suffix == ".c":
+    print(f"Should only be run on a .c file!")
+    exit()
+if "/overlays/actors/" not in str(file_path):
+    print(f"This should only be run on actors, are you sure you gave it the right file?")
+    exit()
+
+map_path = file_path.parent.parent.parent.parent.parent
+map_lines = (map_path / "build" / "mm.map").read_text().splitlines()[10000:]
+
+i = 0
+while not f"{file_path.stem}.o(.text)" in map_lines[i]:
+    i += 1
+i += 2
+assert map_lines[i-1].startswith(" .text")
+
+orig_funcs = set()
+while map_lines[i].startswith("  "):
+    func = map_lines[i].split(" ")[-1]
+    if not (func.endswith("_Init") or func.endswith("_Destroy") or func.endswith("_Update") or func.endswith("_Draw")):
+        orig_funcs.add(func)
+    i += 1
+
+lines = file_path.read_text().splitlines()
+start_func_line = 0
+end_func_line = 0
+func_list = set()
+externs = []
+extern_set = set()
+
+i = 0
+last_extern = 0
+while (not (lines[i].startswith("const") or lines[i].startswith("static") or lines[i].startswith("typedef"))):
+    line = lines[i]
+    
+    if start_func_line == 0 and "(" in line and ");" in line and line.count(" ") == 4:
+        if "_Init(" not in line and "_Destroy(" not in line and "_Update(" not in line and "_Draw(" not in line:
+            start_func_line = i
+    elif line.startswith("extern"):
+        extern_type = line.rsplit(" ",1)[0]
+        extern_name = line.split(" ",2)[2][:-1]
+        if extern_name not in extern_set:
+            externs.append({"type":extern_type, "name":extern_name})
+            extern_set.add(extern_name)
+        if end_func_line == 0:
+            end_func_line = i
+            start_extern_line = i
+        last_extern = i
+    i += 1
+end_extern_line = last_extern + 1
+
+assert start_func_line > 0 and end_func_line > 0
+assert start_extern_line > 0 and end_extern_line > 0
```

Comment:
I pulled this branch to try it and it show me this error
```
$ ./tools/sort_actor_funcs.py src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c 
Traceback (most recent call last):
  File "./tools/sort_actor_funcs.py", line 63, in <module>
    assert start_func_line > 0 and end_func_line > 0
AssertionError
```
I don't really understand what the problem is by just looking at the error. It would be nice if it had a more descriptive error message instead of a plain assert.

Reply:
It's trying to find where you've forward-declared functions already to remove them, needed if directly editing the file but since I removed that and just print them I could remove it.

Reply:
Either removing it or adding a more clear error message would be awesome.